bib/cmd/bootc-image-builder/main.go

```diff
@@ -115,9 +115,19 @@ func makeManifest(c *ManifestConfig, cacheRoot string) (manifest.OSBuildManifest
 	}
 
 	// resolve container
-	resolver := container.NewResolver(c.Architecture.String())
+	hostArch := arch.Current()
+	resolverNative := container.NewResolver(hostArch.String())
+	resolverTarget := container.NewResolver(c.Architecture.String())
```

Comment:
If I'm reading this right, in the standard case, we'll get two identical resolvers and each one will get the same container it will have to resolve.  It's not _terrible_ but a bit wasteful.  Maybe we could instead:
```go
resolvers[hostArch.String()] = container.NewResolver(hostArch.String())
if hostArch.String() != c.Architectyre.String() {
    resolvers[c.Architecture.String()] = container.NewResolver(c.Architecture.String()) 

}

...

if plName == "build" {
    rarch = hostArch.String()
} else {
    rarch = c.Architecture.String()
}
for _, c := range sourceSpecs {
    rarch.Add(c)
}
```

I now also realise that even in this case, we're resolving the same container twice (and have been since we started using the container as a build root).
Again, not terrible, but maybe we can be smarter about it :)

Reply:
Thank you, I will look at this, maybe in a followup as it's not *that* pressing but I agree that it's a bit wasteful

Reply:
Thanks again, I was thinking it could be something like:
```go
index 5e9ed2d..9bb6a5b 100644
--- a/bib/cmd/bootc-image-builder/main.go
+++ b/bib/cmd/bootc-image-builder/main.go
@@ -115,13 +115,18 @@ func makeManifest(c *ManifestConfig, cacheRoot string) (manifest.OSBuildManifest
        }
 
        // resolve container
-       hostArch := arch.Current()
-       resolverNative := container.NewResolver(hostArch.String())
-       resolverTarget := container.NewResolver(c.Architecture.String())
+       hostArch := arch.Current().String()
+       targetArch := c.Architecture.String()
+
+       resolverNative := container.NewResolver(hostArch)
+       resolverTarget := resolverNative
+       if hostArch != targetArch {
+               resolverTarget = container.NewResolver(targetArch)
+       }
 
        containerSpecs := make(map[string][]container.Spec)
        for plName, sourceSpecs := range manifest.GetContainerSourceSpecs() {
-               var resolver container.Resolver
+               var resolver *container.Resolver
                if plName == "build" {
                        resolver = resolverNative
                } else {
```
which requires https://github.com/osbuild/images/pull/433 (so that we get pointers to the resolver first) - I have no super strong opinion here though, happy to discuss more if you think this is less obvious or too verbose :)